src/Ng/ElfieFromCatalogCollector.cs

```diff
@@ -240,6 +190,16 @@ public ElfieFromCatalogCollector(Version indexerVersion, Uri index, NugetService
             {
                 Trace.TraceError(e.ToString());
 
+                WebException webException = e as WebException;
+                if (webException != null && webException.Response != null && ((HttpWebResponse)webException.Response).StatusCode == HttpStatusCode.NotFound)
+                {
+                    // We received a 404 (file not found) when trying to download the file.   
+                    // There's not much we can do here since the package download URL doesn't exist.  
+                    // Return null, which indicates that the package doesn't exist, and continue.  
+                    Trace.TraceError($"The package download URL returned a 404. {package.PackageContent}");
```

Comment:
Yes, retrying here is a good idea. I'll add it. I'll also add a retry in the GetLatestStableVersion(string packageId) method (~line 465.)